src/components/MessagesList/MessagesList.vue

```diff
@@ -272,9 +281,14 @@ export default {
 
 		messagesList: {
 			immediate: true,
-			handler(messages) {
-				const groups = this.prepareMessagesGroups(messages)
-				this.messagesGroupedByAuthor = this.softUpdateMessagesGroups(this.messagesGroupedByAuthor, groups)
+			handler(newMessages, oldMessages) {
+				const newGroups = this.prepareMessagesGroups(newMessages)
+				// Check if messages token has changed or if messages were just loaded
+				if (!!oldMessages || (newMessages.length && newMessages[0].token !== oldMessages[0].token)) {
+					this.messagesGroupedByDateByAuthor = newGroups
+				} else {
+					this.softUpdateByDateGroups(this.messagesGroupedByDateByAuthor, newGroups)
```

Comment:
temporary messages look weird without timestamp. Or it's not provided?
![image](https://github.com/nextcloud/spreed/assets/93392545/de1bc35d-c6f9-4a58-8743-e3317a0521a8)

Reply:
Fixed